nio/client/base_client.py

```diff
@@ -528,6 +528,18 @@ def unignore_device(self, device):
 
         return changed
 
+    def _handle_register(self, response):
+        # type: (Union[LoginResponse, ErrorResponse]) -> None
```

Comment:
This has an incorrect type hint, no? Also do we need separate methods aren't the class attributes the same?

Reply:
the attributes are the same, but doesn't it make more sense for someone reading through the source code for registration to have its own handler? Also helps if matrix API ever changes.

Reply:
Well it isn't that important, you can leave it as is. The API would need to change in quite a breaking way for this to become wrong.